rmgpy/cantherm/explorer.py

```diff
@@ -0,0 +1,236 @@
+###############################################################################
+
+import os
+import numpy as np
+import logging
+import shutil
+
+from rmgpy.rmg.model import CoreEdgeReactionModel
+from rmgpy.rmg.pdep import PDepNetwork
+from rmgpy.molecule.molecule import Molecule
+from rmgpy import settings
+from rmgpy.data.rmg import getDB
+from rmgpy.species import Species
+
+class ExplorerJob(object):
+    def __init__(self, source, pdepjob, explore_tol, energy_tol=np.inf, flux_tol=0.0):
+        self.source = source
+        self.explore_tol = explore_tol
+        self.energy_tol = energy_tol
+        self.flux_tol = flux_tol
+        
+        self.pdepjob = pdepjob
+        
+        if not hasattr(self.pdepjob,'outputFile'):
+            self.pdepjob.outputFile = None
+
+    def copy(self):
+        """
+        Return a copy of the pressure dependence job.
+        """
+        return ExplorerJob(
+            source=self.source,
+            pdepjob=self.pdepjob,
+            explore_tol=self.explore_tol,
+            energy_tol=self.energy_tol,
+            flux_tol=self.flux_tol
+        )
+        
+    def execute(self, outputFile, plot, format='pdf', print_summary=True, speciesList=None, thermoLibrary=None, kineticsLibrary=None):
+        
+        logging.info('Exploring network...')
+        
+        reactionModel = CoreEdgeReactionModel()
+        
+        reactionModel.pressureDependence = self.pdepjob
+        
+        reactionModel.pressureDependence.rmgmode = True
+        
+        if outputFile:
+            reactionModel.pressureDependence.outputFile = os.path.dirname(outputFile)
+        
+        kineticsDatabase = getDB('kinetics')
+        thermoDatabase = getDB('thermo')
+        
+        thermoDatabase.libraries['thermojobs'] = thermoLibrary
+        thermoDatabase.libraryOrder.insert(0,'thermojobs')
+        
+        kineticsDatabase.libraries['kineticsjobs'] = kineticsLibrary
+        kineticsDatabase.libraryOrder.insert(0,('kineticsjobs','Reaction Library'))
+        
+        reactionModel.addSeedMechanismToCore('kineticsjobs')
+        
+        jobRxns = [rxn for rxn in reactionModel.core.reactions]
+        
+        self.jobRxns = jobRxns
+        
+        for lib in kineticsDatabase.libraryOrder:
+            if lib[0] != 'kineticsjobs':
+                reactionModel.addReactionLibraryToEdge(lib[0])
+        
+        if outputFile is not None:
+            if not os.path.exists(os.path.join(reactionModel.pressureDependence.outputFile,'pdep')):
+                os.mkdir(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+            else:
+                shutil.rmtree(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+                os.mkdir(os.path.join(reactionModel.pressureDependence.outputFile,'pdep'))
+            
+        #get the molecular formula for the network
+        
+        mmol = None
+        for spc in self.source:
```

Comment:
Consider using a shorter syntax instead:
```python
if len(self.source) == 1:
    form = self.source[0].molecule[0].getFormula()
else:
    form = self.source[0].molecule[0].merge(self.source[1].molecule[0]).getFormula()
```
Also, the formula can be generated from any isomer instead

Reply:
I feel like the original syntax is shorter.  I think we could conceivably have a network without any isomers.